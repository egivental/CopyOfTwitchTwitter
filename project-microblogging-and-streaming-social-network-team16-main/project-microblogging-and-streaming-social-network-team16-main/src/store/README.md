A short guide on how to use the store in this folder:

## Imports
```
import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
```

## Local State in a Component
```
const [incrementAmount, setIncrementAmount] = useState("2");

```

where `incrementAmount` is the getter and `setIncrementAmount()` is the setter. Use `useState("2")` makes the default value of `incrementAmount` to be `2`.

## Create a Store State
`counterState.js`
```
import { createSlice } from "@reduxjs/toolkit";

export default createSlice({
  name: "counter",
  initialState: {
    value: 0,
  },
  reducers: {
    increment: (state) => {
      // Redux Toolkit allows us to write "mutating" logic in reducers. It
      // doesn't actually mutate the state because it uses the Immer library,
      // which detects changes to a "draft state" and produces a brand new
      // immutable state based on those changes
      state.value += 1;
    },
    decrement: (state) => {
      state.value -= 1;
    },
    incrementByAmount: (state, action) => {
      state.value += action.payload;
    },
  },
});

```

## Import the Counter State
```
import counterState from "./counterState";
```

## Read Store State in a Component
```
const count = useSelector((state) => state.counter.value);
```

## Perform Actions on a Store State in a Component
```
const dispatch = useDispatch();
```
and
```
onClick={() => dispatch(counterState.actions.increment())}
```