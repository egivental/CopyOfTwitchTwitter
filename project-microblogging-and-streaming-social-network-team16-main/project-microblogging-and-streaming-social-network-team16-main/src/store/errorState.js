import { createSlice } from "@reduxjs/toolkit";

const errorState = createSlice({
  name: "error",
  initialState: {
    errors: [],
  },
  reducers: {
    pushError: (state, action) => {
      state.errors.push(action.payload);
    },
    popError: (state) => {
      state.errors.shift();
    },
  },
});

export const { pushError, popError } = errorState.actions;

export default errorState;
