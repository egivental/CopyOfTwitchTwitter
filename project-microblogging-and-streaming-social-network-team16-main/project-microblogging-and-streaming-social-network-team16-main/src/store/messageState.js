import { createSlice } from "@reduxjs/toolkit";
import { apiBaseUrl } from "../constants/constants";
import { pushError } from "./errorState";

const messageState = createSlice({
  name: "message",
  initialState: {
    messages: null,
  },
  reducers: {
    updateMessages: (state, action) => {
      state.messages = action.payload;
    },
  },
});

export const { updateMessages } = messageState.actions;

export const getMessages = (username) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/messages/${username}`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updateMessages(data));
  } else {
    await dispatch(pushError(data.message));
  }
};

export const createMessage = (
  username,
  message,
  attachment,
  attachmentMimeType
) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/messages/${username}`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      message,
      attachment,
      attachment_mime_type: attachmentMimeType,
    }),
  });
  if (response.status === 201) {
    await dispatch(getMessages(username));
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export default messageState;
