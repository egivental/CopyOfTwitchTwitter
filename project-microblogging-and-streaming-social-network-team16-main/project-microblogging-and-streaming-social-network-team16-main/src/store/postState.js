import { createSlice } from "@reduxjs/toolkit";
import { apiBaseUrl } from "../constants/constants";
import { pushError } from "./errorState";

const postState = createSlice({
  name: "post",
  initialState: {
    posts: null,
  },
  reducers: {
    updatePosts: (state, action) => {
      state.posts = action.payload;
    },
  },
});

export const { updatePosts } = postState.actions;

export const getPosts = () => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/posts`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updatePosts(data));
  } else {
    await dispatch(pushError(data.message));
  }
};

export const createPost = (message, attachment, attachmentMimeType) => async (
  dispatch
) => {
  const response = await fetch(`${apiBaseUrl}/post`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      message,
      attachment,
      attachment_mime_type: attachmentMimeType,
    }),
  });
  if (response.status === 201) {
    await dispatch(getPosts());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const deletePost = (postId) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/posts/${postId}`, {
    method: "DELETE",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  if (response.status === 200) {
    await dispatch(getPosts());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const createComment = (postId, message) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/posts/${postId}/comment`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      message,
    }),
  });
  if (response.status === 201) {
    await dispatch(getPosts());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const updateComment = (postId, commentId, message) => async (
  dispatch
) => {
  const response = await fetch(
    `${apiBaseUrl}/posts/${postId}/comments/${commentId}`,
    {
      method: "PUT",
      credentials: "include",
      cache: "no-cache",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        message,
      }),
    }
  );
  if (response.status === 200) {
    await dispatch(getPosts());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const deleteComment = (postId, commentId) => async (dispatch) => {
  const response = await fetch(
    `${apiBaseUrl}/posts/${postId}/comments/${commentId}`,
    {
      method: "DELETE",
      credentials: "include",
      cache: "no-cache",
      headers: {
        "Content-Type": "application/json",
      },
    }
  );
  if (response.status === 200) {
    await dispatch(getPosts());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export default postState;
