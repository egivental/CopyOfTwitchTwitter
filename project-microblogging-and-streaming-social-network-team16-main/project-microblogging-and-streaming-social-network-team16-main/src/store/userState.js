import { createSlice } from "@reduxjs/toolkit";
import { apiBaseUrl } from "../constants/constants";
import { pushError } from "./errorState";

const userState = createSlice({
  name: "user",
  initialState: {
    isUserLoggedIn: null,
    user: null,
    followers: [],
    followees: [],
    otherUsers: [],
  },
  reducers: {
    updateUser: (state, action) => {
      state.user = action.payload;
      if (state.user === null) {
        state.isUserLoggedIn = false;
        state.followers = [];
        state.followees = [];
        state.otherUsers = [];
      } else {
        state.isUserLoggedIn = true;
      }
    },
    updateFollowers: (state, action) => {
      state.followers = action.payload;
    },
    updateFollowees: (state, action) => {
      state.followees = action.payload;
    },
    updateOtherUsers: (state, action) => {
      state.otherUsers = action.payload;
    },
  },
});

export const {
  updateUser,
  updateFollowers,
  updateFollowees,
  updateOtherUsers,
} = userState.actions;

export const createUser = (username, password) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/user`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      username,
      password,
      photo: null,
      photo_mime_type: null,
    }),
  });
  if (response.status === 201) {
    await dispatch(getSession());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const getUsers = () => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/users`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updateOtherUsers(data));
  } else {
    await dispatch(pushError(data.message));
  }
};

export const getUser = (username) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/users/${username}`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updateUser(data));
  } else {
    await dispatch(pushError(data.message));
    await dispatch(updateUser(null));
  }
};

export const updateUserProfile = (
  username,
  password,
  photo,
  photoMimeType
) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/users/${username}`, {
    method: "PUT",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      password,
      photo,
      photo_mime_type: photoMimeType,
    }),
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(getUser(username));
    alert("User profile successfully updated!");
  } else {
    await dispatch(pushError(data.message));
  }
};

export const deleteUser = (username) => async (dispatch, getState) => {
  const state = getState();
  const loggedInUsername = state.user.user.username;
  const response = await fetch(`${apiBaseUrl}/users/${username}`, {
    method: "DELETE",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    if (username === loggedInUsername) {
      // Deactivated own account
      await dispatch(deleteSession());
    } else {
      // Blocked another user, refresh followers, followees, and other users
      await dispatch(getUsers());
      await dispatch(getFollowers(username));
      await dispatch(getFollowees(username));
    }
  } else {
    await dispatch(pushError(data.message));
  }
};

export const getFollowers = (username) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/users/${username}/followers`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updateFollowers(data));
  } else {
    await dispatch(pushError(data.message));
  }
};

export const getFollowees = (username) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/users/${username}/followees`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updateFollowees(data));
  } else {
    await dispatch(pushError(data.message));
  }
};

export const followUser = (username, followeeUsername) => async (dispatch) => {
  const response = await fetch(
    `${apiBaseUrl}/users/${username}/followees/${followeeUsername}`,
    {
      method: "PUT",
      credentials: "include",
      cache: "no-cache",
      headers: {
        "Content-Type": "application/json",
      },
    }
  );
  if (response.status === 200) {
    await dispatch(getFollowees(username));
    await dispatch(getUsers());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const unfollowUser = (username, followeeUsername) => async (
  dispatch
) => {
  const response = await fetch(
    `${apiBaseUrl}/users/${username}/followees/${followeeUsername}`,
    {
      method: "DELETE",
      credentials: "include",
      cache: "no-cache",
      headers: {
        "Content-Type": "application/json",
      },
    }
  );
  if (response.status === 200) {
    await dispatch(getFollowees(username));
    await dispatch(getUsers());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const getSession = () => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/session`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(getUser(data.username));
  } else {
    await dispatch(updateUser(null));
  }
};

export const createSession = (username, password) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/session`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      username,
      password,
    }),
  });
  if (response.status === 201) {
    await dispatch(getUser(username));
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const deleteSession = () => async (dispatch) => {
  await fetch(`${apiBaseUrl}/session`, {
    method: "DELETE",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  await dispatch(getSession());
};

export const getMediaUrl = (id) => `${apiBaseUrl}/media/${id}`;

export default userState;
