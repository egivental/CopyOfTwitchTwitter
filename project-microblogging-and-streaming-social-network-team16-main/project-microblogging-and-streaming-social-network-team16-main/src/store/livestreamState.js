import { createSlice } from "@reduxjs/toolkit";
import { apiBaseUrl } from "../constants/constants";
import { pushError } from "./errorState";
import { getSession } from "./userState";

const livestreamState = createSlice({
  name: "livestream",
  initialState: {
    livestreams: null,
  },
  reducers: {
    updateLivestreams: (state, action) => {
      state.livestreams = action.payload;
    },
  },
});

export const { updateLivestreams } = livestreamState.actions;

export const createLivestream = () => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/livestream`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  if (response.status === 201) {
    await dispatch(getLivestreams());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const getLivestreams = () => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/livestreams`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updateLivestreams(data));
    await dispatch(getSession());
  } else {
    await dispatch(pushError(data.message));
  }
};

export const deleteLivestream = (room) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/livestreams/${room}`, {
    method: "DELETE",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  if (response.status === 200) {
    await dispatch(getLivestreams());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const createLivestreamComment = (room, message) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/livestreams/${room}/comment`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      message,
    }),
  });
  if (response.status === 201) {
    await dispatch(getLivestreams());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const updateLivestreamComment = (room, commentId, message) => async (
  dispatch
) => {
  const response = await fetch(
    `${apiBaseUrl}/livestreams/${room}/comments/${commentId}`,
    {
      method: "PUT",
      credentials: "include",
      cache: "no-cache",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        message,
      }),
    }
  );
  if (response.status === 200) {
    await dispatch(getLivestreams());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const deleteLivestreamComment = (room, commentId) => async (
  dispatch
) => {
  const response = await fetch(
    `${apiBaseUrl}/livestreams/${room}/comments/${commentId}`,
    {
      method: "DELETE",
      credentials: "include",
      cache: "no-cache",
      headers: {
        "Content-Type": "application/json",
      },
    }
  );
  if (response.status === 200) {
    await dispatch(getLivestreams());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export default livestreamState;
