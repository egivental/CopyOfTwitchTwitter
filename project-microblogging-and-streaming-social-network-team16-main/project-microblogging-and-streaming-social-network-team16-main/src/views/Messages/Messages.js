import _ from "lodash";
import React, { useState, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import Image from "react-bootstrap/Image";
import Card from "react-bootstrap/Card";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Dropdown from "react-bootstrap/Dropdown";
import moment from "moment";
import { createMessage, getMessages } from "../../store/messageState";
import { getFollowers, getFollowees, getMediaUrl } from "../../store/userState";

import ContentComposer from "../../components/ContentComposer/ContentComposer";

/**
 * @returns {Messages} A Messages React component.
 */
function Messages() {
  const dispatch = useDispatch();
  const messageState = useSelector((state) => state.message);
  const userState = useSelector((state) => state.user);
  const [initialContactsFetched, setInitialContactsFetched] = useState(false);
  const [otherUser, setOtherUser] = useState(null);

  // Fetch contacts
  if (!initialContactsFetched) {
    if (userState.user) {
      setInitialContactsFetched(true);
      dispatch(getFollowers(userState.user.username));
      dispatch(getFollowees(userState.user.username));
    }
  }

  // Get contacts
  const contacts = _.uniq(
    userState.followers
      .map((u) => u.username)
      .concat(userState.followees.map((u) => u.username))
  );

  // Periodically update messages
  useEffect(() => {
    const intervalId = setInterval(() => {
      dispatch(getMessages(otherUser));
    }, 1500);
    return () => {
      clearInterval(intervalId);
    };
  }, [dispatch, otherUser]);

  let { messages } = messageState;

  // Only fetch messages if messages haven't been fetched yet (prevents an infinite loop)
  if (messages === null) {
    dispatch(getMessages(otherUser));
    messages = [];
  }

  // Render messages
  const displayMessages = messages.map((message) => (
    <Card className="mb-4" key={message.id}>
      <Row>
        <Col md="auto">
          <Image
            width={80}
            height={80}
            alt="80x80"
            className="m-3"
            src={
              message.photo
                ? getMediaUrl(message.photo)
                : "https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png"
            }
          />
        </Col>
        <Col>
          <Card.Body>
            <Card.Title>@{message.from_user}</Card.Title>
            <div>
              <Card.Text>{message.message}</Card.Text>
            </div>
            {message.attachment_type && (
              <Row>
                <Col
                  style={{
                    marginTop: "20px",
                    textAlign: "center",
                  }}
                >
                  {message.attachment_type &&
                  message.attachment_type.startsWith("image") ? (
                    <img
                      src={getMediaUrl(message.attachment)}
                      alt={message.message}
                      style={{ maxWidth: "60%" }}
                    />
                  ) : null}
                  {message.attachment_type &&
                  message.attachment_type.startsWith("video") ? (
                    <video
                      src={getMediaUrl(message.attachment)}
                      style={{ maxWidth: "60%" }}
                      controls
                      autoPlay
                      loop
                    />
                  ) : null}
                  {message.attachment_type &&
                  message.attachment_type.startsWith("audio") ? (
                    <audio
                      src={getMediaUrl(message.attachment)}
                      style={{ maxWidth: "60%" }}
                      controls
                      loop
                    />
                  ) : null}
                </Col>
              </Row>
            )}
          </Card.Body>
        </Col>
      </Row>
      <Card.Footer className="text-muted">
        <Row>
          <Col>{moment.unix(message.date / 1000).fromNow()}</Col>
          <Col md="auto">
            {userState.user.username === message.from_user ? (
              <span>
                <b>Delivered Date: </b>
                {message.delivered_date === 0
                  ? "TBD"
                  : moment.unix(message.delivered_date / 1000).fromNow()}{" "}
                <b style={{ marginLeft: "15px" }}>Read Date: </b>
                {message.read_date === 0
                  ? "TBD"
                  : moment.unix(message.read_date / 1000).fromNow()}
              </span>
            ) : null}
          </Col>
        </Row>
      </Card.Footer>
    </Card>
  ));

  // Render
  return (
    <div>
      <div id="title" style={{ marginBottom: "40px" }}>
        <h1>Messages</h1>
      </div>
      Select a contact to direct message:{" "}
      <Dropdown
        style={{ marginBottom: "20px" }}
        onSelect={(e) => setOtherUser(e)}
      >
        <Dropdown.Toggle variant="secondary" id="dropdown-basic">
          {otherUser || "None"}
        </Dropdown.Toggle>
        <Dropdown.Menu>
          {contacts.map((c) => (
            <Dropdown.Item key={`dropdown-${c}`} eventKey={c}>
              @{c}
            </Dropdown.Item>
          ))}
        </Dropdown.Menu>
      </Dropdown>
      {otherUser ? (
        <div>
          <ContentComposer
            action="Send Message"
            title={`Send a direct message to @${otherUser}...`}
            createAction={(message, attachment, attachmentType) =>
              createMessage(otherUser, message, attachment, attachmentType)
            }
          />
          <h5>Messages between you and {`@${otherUser}`}</h5>
          <hr />
          <div>{displayMessages}</div>
          {displayMessages.length === 0 ? "No messages." : null}
        </div>
      ) : (
        <div>
          <hr />
          No user selected to message with.
        </div>
      )}
    </div>
  );
}

export default Messages;
