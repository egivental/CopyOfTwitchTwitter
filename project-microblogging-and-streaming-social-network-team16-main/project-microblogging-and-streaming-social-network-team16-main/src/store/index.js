import { configureStore } from "@reduxjs/toolkit";
import errorState from "./errorState";
import messageState from "./messageState";
import postState from "./postState";
import userState from "./userState";
import livestreamState from "./livestreamState";

export default configureStore({
  reducer: {
    error: errorState.reducer,
    user: userState.reducer,
    post: postState.reducer,
    message: messageState.reducer,
    livestream: livestreamState.reducer,
  },
});
