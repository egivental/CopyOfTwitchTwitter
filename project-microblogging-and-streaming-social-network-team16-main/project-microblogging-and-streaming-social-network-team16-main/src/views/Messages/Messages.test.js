import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../store/index";
import Messages from "./Messages";

test("renders page", () => {
  const { getByText } = render(
    <Provider store={store}>
      <Messages />
    </Provider>
  );

  expect(getByText(/Select a contact to direct message/i)).toBeInTheDocument();
  expect(getByText(/No user selected to message with/i)).toBeInTheDocument();
});

// snapshot testing
test("Messages matches snapshot", () => {
  const component = renderer.create(
    <Provider store={store}>
      <Messages />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
