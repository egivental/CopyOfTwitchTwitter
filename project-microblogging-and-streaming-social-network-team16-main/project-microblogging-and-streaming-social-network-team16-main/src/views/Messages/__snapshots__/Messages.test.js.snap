// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages matches snapshot 1`] = `
<div>
  <div
    id="title"
    style={
      Object {
        "marginBottom": "40px",
      }
    }
  >
    <h1>
      Messages
    </h1>
  </div>
  Select a contact to direct message:
   
  <div
    className="dropdown"
    onKeyDown={[Function]}
    style={
      Object {
        "marginBottom": "20px",
      }
    }
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-secondary"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      None
    </button>
  </div>
  <div>
    <hr />
    No user selected to message with.
  </div>
</div>
`;
